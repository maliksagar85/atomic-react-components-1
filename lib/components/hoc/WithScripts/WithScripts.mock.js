export const defaultConfig = [
  {
    src: 'https://maps.googleapis.com/maps/api/js',
    async: true,
    scriptId: 'maps1',
  },
  {
    src: 'https://pay.google.com/gp/p/js/pay.js',
    scriptId: 'pay1',
  },
];

export const ignoreDuplicateScriptID = [
  {
    src: 'https://maps.googleapis.com/maps/api/js?ignoreDups=true',
    async: true,
    scriptId: 'maps2',
  },
  {
    src: 'https://pay.google.com/gp/p/js/pay.js?ignoreDups=true',
    scriptId: 'pay2',
  },
  {
    src: 'https://pay.google.com/gp/p/js/pay.js?ignoreDups=true',
    scriptId: 'pay2',
  },
];
