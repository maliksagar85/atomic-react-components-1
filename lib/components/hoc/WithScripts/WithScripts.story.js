import React from 'react';
import { storiesOf } from '@storybook/react';
import { defaultConfig, ignoreDuplicateScriptID } from './WithScripts.mock';

// Import Styled Component to showcase variations
import withScripts from './WithScripts';

const comp1 = () => (
  <div>
    <code>
      src: https://maps.googleapis.com/maps/api/js?ignoreDups=true, async: true, scriptId: maps2
      <br />
      src: https://pay.google.com/gp/p/js/pay.js?ignoreDups=true, scriptId: pay2,
      <br />
    </code>
    Hoc will consume the wrapper Component A and injected respective script details
  </div>
);

const comp2 = () => (
  <div>
    <code>
      src: https://maps.googleapis.com/maps/api/js?ignoreDups=true, async: true, scriptId: maps2
      <br />
      src: https://pay.google.com/gp/p/js/pay.js?ignoreDups=true, scriptId: pay2,
      <br />
      src: https://pay.google.com/gp/p/js/pay.js?ignoreDups=true, scriptId: pay2,
      <br />
    </code>
    Hoc will consume the wrapper Component B and injected respective script details while ignoring
    duplicate scriptIDs Check network tab and filter maps
  </div>
);

const ComponentA = withScripts(comp1, defaultConfig);
const ComponentB = withScripts(comp2, ignoreDuplicateScriptID);

// eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('HOC/withScript', module)
  .addParameters({ jest: ['ScriptInjection', 'ScriptInjectionVanilla'] })
  .add('WithScripts A', () => <ComponentA />)
  .add('WithScripts No Duplicate Scripts', () => <ComponentB />);
