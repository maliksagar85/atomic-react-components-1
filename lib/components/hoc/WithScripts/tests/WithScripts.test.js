import { mount } from 'enzyme';

import withScripts from '../index';

import { defaultConfig, ignoreDuplicateScriptID } from '../WithScripts.mock';

const comp1 = () => (
  <div>Hoc will consume the wrapper Component A and injected respective script details</div>
);

const ComponentA = withScripts(comp1, defaultConfig);

const comp2 = () => (
  <div>
    Hoc will consume the wrapper Component A and injected respective script details without
    Duplicate ids
  </div>
);

const ComponentB = withScripts(comp2, ignoreDuplicateScriptID);

describe('<WithScripts />', () => {
  beforeEach(() => {
    // Avoid `attachTo: document.body` Warning
    const div = document.createElement('div');
    div.setAttribute('id', 'container');
    document.body.appendChild(div);
  });

  afterEach(() => {
    const div = document.getElementById('container');
    if (div) {
      document.body.removeChild(div);
    }
    const scriptsCollection = document.getElementsByTagName('script');
    while (scriptsCollection.length > 0) {
      scriptsCollection[0].parentNode.removeChild(scriptsCollection[0]);
    }
  });

  test('should render correctly', () => {
    const WithScriptsComponent = mount(<ComponentA />, {
      attachTo: document.getElementById('container'),
    });
    expect(document.getElementsByTagName('script').length).toEqual(2);
    expect(WithScriptsComponent).toMatchSnapshot();
    WithScriptsComponent.detach();
  });
  test('should not render duplicate ids', () => {
    const WithScriptsComponent = mount(<ComponentB />, {
      attachTo: document.getElementById('container'),
    });
    expect(document.getElementsByTagName('script').length).toEqual(2);
    expect(WithScriptsComponent).toMatchSnapshot();
    WithScriptsComponent.detach();
  });
});
