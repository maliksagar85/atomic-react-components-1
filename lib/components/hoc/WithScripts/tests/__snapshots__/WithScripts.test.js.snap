// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithScripts /> should not render duplicate ids 1`] = `
<WithScripts(comp2)>
  <comp2
    injectedProp={Object {}}
  >
    <div>
      Hoc will consume the wrapper Component A and injected respective script details without Duplicate ids
    </div>
  </comp2>
</WithScripts(comp2)>
`;

exports[`<WithScripts /> should render correctly 1`] = `
<WithScripts(comp1)>
  <comp1
    injectedProp={Object {}}
  >
    <div>
      Hoc will consume the wrapper Component A and injected respective script details
    </div>
  </comp1>
</WithScripts(comp1)>
`;
